src/observer.ts

```diff
@@ -474,6 +468,22 @@ export class Observer {
       height: epochStartHeight,
     });
 
+    this.referenceGatewayResolutionCache = new ReadThroughPromiseCache<
+      string,
+      ArnsResolution
+    >({
+      cacheParams: {
+        cacheCapacity: prescribedNames.length + chosenNames.length,
+        cacheTTL: 5 * 60_000,
+      },
+      readThroughFunction: async (name: string) =>
+        getArnsResolution({
+          url: `https://${name}.${this.referenceGatewayHost}/`,
+          got: this.gotClient,
+          entropy,
+        }),
+    });
+
```

Comment:
i rec the cache above when you create the cache attribute and use dynamic cache keys similar to here - https://github.com/ar-io/arns-service/blob/main/src/api/warp.ts#L57-L91

we then fetch from the cache using the caheKey [here](https://github.com/ar-io/arns-service/blob/main/src/api/warp.ts#L333-L369)

Reply:
example

```
class ArNSCacheKey {
  constructor(
    public readonly name: string;
    public readonly host: string;
  ) {}

  toString(): string {
    return `https://${this.name}.${this.host});
  }

  // Facilitate ReadThroughPromiseCache key derivation
  toJSON() {
    return { cacheKey: this.toString() };
  }
}
```
and instantiate the cache at the top of the cache calling our internal promise
```
protected referenceGatewayResolutionCache = new ReadThroughPromiseCache<
      string,
      ArnsResolution
    >({
      cacheParams: {
        cacheCapacity: prescribedNames.length + chosenNames.length,
        cacheTTL: 5 * 60_000,
      },
      readThroughFunction: async (arnsCacheKey: ArNSCacheKey) =>
        getArnsResolution({
          url: arnsCacheKey.toString()
          got: this.gotClient,
          entropy,
        }),
    });
```

Reply:
We can use arns name for cache key here as we’re only caching reference gateway resolution.

Reply:
Yeah, cool technique, but in this case we only have one host.